pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -1,5 +1,18 @@
 package v1alpha5
 
+// SetClusterConfigDefaults will set defaults for a given cluster
+func SetClusterConfigDefaults(cfg *ClusterConfig) error {
+	if cfg.HasClusterCloudWatchLogging() {
+		for _, logType := range cfg.CloudWatch.ClusterLogging.EnableTypes {
+			switch logType {
+			case "all", "*":
```

Comment:
Since we don't validate other log types here, this makes it possible to pass invalid log types and bypass the validation in `ValidateClusterConfig` as the value is reset to `SupportedCloudWatchClusterLogTypes()` . 

So this would pass validation:
```yaml
enableTypes: ["all", "asdf", "asdfasdf"]
```

I think we should only support "all" or "*" when `enableTypes` is a single-element array.


```golang
func SetClusterConfigDefaults(cfg *ClusterConfig) {
	if cfg.HasClusterCloudWatchLogging() && len(cfg.CloudWatch.ClusterLogging.EnableTypes) == 1 {
		switch cfg.CloudWatch.ClusterLogging.EnableTypes[0] {
		case "all", "*":
			cfg.CloudWatch.ClusterLogging.EnableTypes = SupportedCloudWatchClusterLogTypes()
		}
	}
}
```

Reply:
Yes, but `enableTypes: ["all", "api", "audit"]` should be valid. Perhaps validation should run before defaulting?

Reply:
I think we could just open an issue to address this later.

Reply:
>Yes, but enableTypes: ["all", "api", "audit"] should be valid

Not sure if we should allow adding other types when `all` or `*` is present.